src/test/java/io/sniffy/socket/SnifferSocketImplFactoryTest.java

```diff
@@ -0,0 +1,161 @@
+package io.sniffy.socket;
+
+import io.sniffy.Sniffer;
+import io.sniffy.Spy;
+import io.sniffy.Threads;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.*;
+
+public class SnifferSocketImplFactoryTest {
+
+    private final static byte[] RESPONSE = new byte[]{9,8,7,6,5,4,3,2};
+    private final static byte[] REQUEST = new byte[]{1, 2, 3, 4};
+
+    private static InetAddress localhost;
+
+    @BeforeClass
+    public static void resolveLocalhost() throws UnknownHostException {
+        localhost = InetAddress.getByName(null);
+    }
+
+    @Rule
+    public EchoServerRule echoServerRule = new EchoServerRule(RESPONSE);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/sniffy/sniffy/file/2296812001/issues/source?bid=3005960&fileBranchId=3034483#l34)